### PR TITLE
chore(generators): send warning regarding invalid template to stderr

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What kind of change does this PR introduce?**
chore

**Did you add tests for your changes?**
Update test suite.

**If relevant, did you update the documentation?**
N/A

**Summary**
https://github.com/webpack/webpack-cli/blob/af6c02b130a35b62b94542680aa7ea41561e77d7/packages/generators/src/addon-generator.ts#L60

**Does this PR introduce a breaking change?**
Nope

**Other information**
N/A

### CODE REVIEW

File: packages/generators/src/addon-generator.ts [hunk 1]
```diff
@@ -57,7 +58,7 @@ const addonGenerator = (
 
         public async prompting(): Promise<void> {
             if (!this.supportedTemplates.includes(this.template)) {
-                this.utils.logger.warn(`⚠ ${this.template} is not a valid template, please select one from below`);
+                this.utils.logger.log(`${yellow(`⚠ ${this.template} is not a valid template, please select one from below`)}`);
```
Comment (alexander-akait) [inline packages/generators/src/addon-generator.ts:61]: hm, why we need change warn on log?

Comment (jamesgeorge007) [reply packages/generators/src/addon-generator.ts:61]: We're writing to `stdout` for the case of `init`.
https://github.com/webpack/webpack-cli/blob/af6c02b130a35b62b94542680aa7ea41561e77d7/packages/generators/src/init-generator.ts#L61

We follow a similar implementation for the `loader` and `plugin` subcommands. However, using `logger.warn()` writes to `stdin`.
https://github.com/webpack/webpack-cli/blob/af6c02b130a35b62b94542680aa7ea41561e77d7/packages/generators/src/addon-generator.ts#L60

Comment (evenstensberg) [reply packages/generators/src/addon-generator.ts:61]: writing to stderr is better

**Review by alexander-akait — APPROVED** [review]

**Review by ddsmatt — COMMENTED** [review]
Lots of changes without explicit approval. Thanks.

**Pull request merged**